meson.build

```diff
@@ -0,0 +1,119 @@
+project(
+  'libdicom',
+  'c',
+  license: 'MIT',
+  meson_version : '>=0.54',
+  version: '0.1.0',
+)
+
+# dependencies
+system_uthash = get_option('system_uthash')
+if system_uthash.enabled() or system_uthash.auto()
+  cc = meson.get_compiler('c')
+  found_uthash = cc.check_header(
+    'utarray.h',
+    required : system_uthash
+  )
+  found_uthash = found_uthash and cc.check_header(
+    'uthash.h',
+    required : system_uthash
+  )
+  if found_uthash
+    if system_uthash.auto()
+      message('Using system uthash')
+    endif
+    uthash = ''
+  else
+    message('Using bundled uthash')
+    uthash = 'uthash'
+  endif
+else
+  uthash = 'uthash'
+endif
+check = dependency(
+  'check',
+  required : false,
+  version : '>=0.9.6',
+)
+
+# options
+cfg = configuration_data()
+suffixed_version = meson.project_version()
+version_suffix = get_option('version_suffix')
+if version_suffix != ''
+  suffixed_version += '-' + version_suffix
+endif
+cfg.set(
+  'HAVE_CHECK',
+  check.found(),
+  description : 'Define if the Check package is available.',
+)
+cfg.set_quoted(
+  'SRCDIR',
+  meson.current_source_dir().replace('\\', '\\\\'),
+  description : 'Define to the path to the source directory.',
+)
+cfg.set_quoted(
+  'SUFFIXED_VERSION',
+  suffixed_version,
+  description : 'The package version string including any suffix.',
+)
+configure_file(
+  output : 'config.h',
+  configuration : cfg,
+)
+
+# libdicom
+lib_sources = [
+  'src/dicom.c',
+  'src/dicom-data.c',
+  'src/dicom-dict.c',
+  'src/dicom-file.c',
+]
+install_headers(
+  'src/dicom.h',
+  subdir : 'dicom',
+)
+libdicom = library(
+  'dicom',
+  lib_sources,
+  c_args : ['-Wextra', '-Wformat', '-Wformat-security'],
```

Comment:
Do these work on clang too? It migh be safest to generate extra c args with eg.:

```
extra_args = []
if meson.get_compiler('c').get_id() == 'gcc'
  extra_args += [
    '-Wall',
    '-Werror',
    '-Wextra',
    '-Wstrict-prototypes',
    '-Wmissing-prototypes',
    '-Wmissing-declarations',
    '-Wnested-externs',
    '-Wformat',
    '-Wformat-security'
  ]
endif
```

Reply:
Yeah, they do, but MSVC could be an issue.  It turns out that Meson knows about "GCC-like" compilers, so I'll condition on that instead.  `-Wall` is enabled automatically and `-Werror` tends to break the build on future compilers if enabled by default, but I'll add the others.

Reply:
Ah, it looks like the right way to set `-Werror` is setting `warning_level` to 2.

Reply:
Ah, it looks like the right way to set `-Wextra` is setting `warning_level` to 2.